syntax = "proto3";

package darmok.protobuf;

import "varying.proto";
import "program.proto";
import "material.proto";
import "light.proto";

message AssimpSceneImportConfig {
    oneof program {
        string program_path = 1;
        StandardProgram.Type standard_program = 2;
    }
    repeated string program_defines = 3;
    bool embed_textures = 4;
    string default_texture_path = 5;
    repeated string skip_meshes_regex = 6;
    repeated string skip_nodes_regex = 7;
    string root_mesh_regex = 8;
    optional Material.OpacityType opacity = 9;
    optional Light.ShadowType shadow_type = 10;
}